modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -226,13 +238,25 @@ func (k Keeper) SetInterchainAccountAddress(ctx sdk.Context, connectionID, portI
 // IsMiddlewareEnabled returns true if the underlying application callbacks are enabled for given port and connection identifier pair, otherwise false
 func (k Keeper) IsMiddlewareEnabled(ctx sdk.Context, portID, connectionID string) bool {
 	store := ctx.KVStore(k.storeKey)
-	return store.Has(icatypes.KeyIsMiddlewareEnabled(portID, connectionID))
+	return bytes.Equal(icatypes.MiddlewareEnabled, store.Get(icatypes.KeyIsMiddlewareEnabled(portID, connectionID)))
+}
+
+// IsMiddlewareDisabled returns true if the underlying application callbacks are disabled for the given port and connection identifier pair, otherwise false
+func (k Keeper) IsMiddlewareDisabled(ctx sdk.Context, portID, connectionID string) bool {
+	store := ctx.KVStore(k.storeKey)
+	return bytes.Equal(icatypes.MiddlewareDisabled, store.Get(icatypes.KeyIsMiddlewareEnabled(portID, connectionID)))
 }
 
 // SetMiddlewareEnabled stores a flag to indicate that the underlying application callbacks should be enabled for the given port and connection identifier pair
 func (k Keeper) SetMiddlewareEnabled(ctx sdk.Context, portID, connectionID string) {
 	store := ctx.KVStore(k.storeKey)
-	store.Set(icatypes.KeyIsMiddlewareEnabled(portID, connectionID), []byte{byte(1)})
+	store.Set(icatypes.KeyIsMiddlewareEnabled(portID, connectionID), icatypes.MiddlewareEnabled)
+}
+
+// SetMiddlewareDisabled stores a flag to indicate that the underlying application callbacks should be disabled for the given port and connection identifier pair
```

Comment:
Sorry, but I didn't have time to review the original PR, so I am just mentioning this as a suggestion that could be taken up in a follow up PR if you think it's useful. It's more of a code hygiene thing, anyway...

Instead of having two separate functions (`SetMiddlewareEnabled` and `SetMiddlewareDisabled`), would it be possible instead to have just one (e.g. `SetMiddlewareStatus) and pass as third argument either `icatypes.MiddlewareEnabled` or `icatypes.MiddlewareDisabled`? The key could be also `icatypes.KeyIsMiddlewareStatus` instead `icatypes.KeyIsMiddlewareEnabled`.

Reply:
I thought about this and originally when I was working on the issue I had one `IsMiddlewareEnabled()` func which retruned a true/false boolean value. 

The problem with this is you need to handle more than just two cases cases. Middleware is either enabled, disabled or hasn't been set. In the case of it not being set it would be the first time a handshake is initiated for that connection/owner tuple and we need to allow it proceed these checks. 

I think with your suggestion of `SetMiddlewareStatus` you'd also need a `CheckMiddlewareStatus(enabled/disabled)` which takes an arg and returns true/false in the appropriate cases. I'm sure there's a few different approachs you could take here. I think the current code and logic is pretty readable when you consider all the scenarios.